src/module.c

```diff
@@ -13489,6 +13492,111 @@ int RM_GetDbIdFromDefragCtx(RedisModuleDefragCtx *ctx) {
+ *
+ * In order to know the number of elements were added, the additional argument
+ * 'added' must be passed, that populates the integer by reference
+ * setting it to the number of elements added depending on the outcome of the operation.
+ * The 'added' argument can be NULL if the caller is not interested
+ * to know if the number of elements were really added.
+ *
+ * Empty keys will be created with set key type and continue. */
+int RM_SetAdd(RedisModuleKey *key, RedisModuleString **eles, size_t numeles, size_t *added) {
+    if (!(key->mode & REDISMODULE_WRITE)) return REDISMODULE_ERR;
+    if (key->value && key->value->type != OBJ_SET) return REDISMODULE_ERR;
+    if (key->value == NULL) moduleCreateEmptyKey(key,REDISMODULE_KEYTYPE_SET);
+    size_t i, numadded = 0;
+    for (i = 0; i < numeles; i++) {
+        numadded += setTypeAdd(key->value,eles[i]->ptr);
+    }
+    if (added) *added = numadded;
+
+    return REDISMODULE_OK;
+}
+
+/* Remove the specified element from the set and the key will be
+ * removed if has no any element in after remove elements operation.
+ * The function returns REDISMODULE_OK on success, and REDISMODULE_ERR
+ * on one of the following conditions:
+ *
+ * * The key was not opened for writing.
+ * * The key is of the wrong type.
+ *
+ * Key will be removed if has no any element in after remove elements operation
+ *
+ * The return value does NOT indicate the fact the element was really
+ * removed (since it existed) or not, just if the function was executed
+ * with success.
+ *
+ * In order to know the number of elements were removed, the additional argument
+ * 'deleted' must be passed, that populates the integer by reference
+ * setting it to the number of elements removed depending on the outcome of the operation.
+ * The 'deleted' argument can be NULL if the caller is not interested
+ * to know if the number of elements were really removed.
+ *
+ * Empty keys will be handled correctly by doing nothing. */
+int RM_SetRem(RedisModuleKey *key, RedisModuleString **eles, size_t numeles, size_t *deleted) {
+    if (!(key->mode & REDISMODULE_WRITE)) return REDISMODULE_ERR;
+    if (key->value && key->value->type != OBJ_SET) return REDISMODULE_ERR;
+
+    size_t i, numdeleted = 0;
+    for (i = 0; i < numeles; i++) {
+        if (key->value != NULL) {
+            numdeleted += setTypeRemove(key->value,eles[i]->ptr);
+            moduleDelKeyIfEmpty(key);
+        }
+    }
+    if (deleted) *deleted = numdeleted;
+    return REDISMODULE_OK;
+}
+
+/* Return 1 as member of the key or 0 as not member of the key. */
+int RM_SetIsMember(RedisModuleKey *key, RedisModuleString *ele) {
+    if (!key->value || key->value->type != OBJ_SET) return 0;
+    return setTypeIsMember(key->value,ele->ptr);
+}
+
+/* Return the set cardinality (number of elements) of the set stored at key. */
+int RM_SetCard(RedisModuleKey *key) {
+    if (!key->value || key->value->type != OBJ_SET) return 0;
+    return setTypeSize(key->value);
+}
```

Comment:
why not just rely on RM_ValueLength?

Reply:
You are right. The idea was to map set command one by one. but since there is common API to get the same functionality. This function could be removed. I'll remove it.